src/Diagnostics.ModelsAndUtils/Models/Resource/ArmResource.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Diagnostics.ModelsAndUtils.Attributes;
+using Diagnostics.ModelsAndUtils.ScriptUtilities;
+
+namespace Diagnostics.ModelsAndUtils.Models
+{
+    /// <summary>
+    /// Resource representing a Generic Arm Resource
+    /// </summary>
+    public class ArmResource : ArmResourceFilter, IResource
+    {
+        public string SubscriptionId { get; set; }
+        public string ResourceGroup { get; set; }
+        public string Name { get; set; }
+
+        public string ResourceUri
+        {
+            get
+            {
+                return UriUtilities.BuildAzureResourceUri(SubscriptionId, ResourceGroup, Name, Provider, ResourceTypeName);
+            }
+        }
+
+        public bool IsApplicable(IResourceFilter filter)
+        {
+            return filter is ArmResource;
```

Comment:
you can do the Provider Type and Resource Type string matching also here.

Reply:
Makes sense, updated